cmd/aws.go

```diff
@@ -98,10 +113,36 @@ func init() {
 	awsCmd.AddCommand(awsResourcesCmd)
 
 	// Required flags
-	awsCmd.Flags().String("access-key", "", "Access Key (required)")
-	awsCmd.Flags().String("secret-key", "", "Secret Key (required)")
-	awsCmd.Flags().String("region", "", "Region to search in, for now * it's not supported (required)")
+	awsCmd.Flags().String("aws-access-key", "", "Access Key (required)")
+	awsCmd.Flags().String("aws-secret-access-key", "", "Secret Key (required)")
+	awsCmd.Flags().String("aws-default-region", "", "Region to search in, for now * it's not supported (required)")
+	awsCmd.Flags().String("aws-shared-credentials-file", "", "Path to the AWS credential path")
+	awsCmd.Flags().String("aws-profile", "", "Name of the Profile to use with the Credentials")
 
 	// Filter flags
 	awsCmd.Flags().StringSliceVarP(&tags, "tags", "t", []string{}, "List of tags to filter with format 'NAME:VALUE'")
 }
+
+// loadAWSCredentials will first read from ENV and if AccessKey and SecretAccessKey are not found (both of them)
+// will fallback to the SharedCredentials with the profile
+func loadAWSCredentials() error {
+	creds := credentials.NewCredentials(&credentials.ChainProvider{
+		Providers: []credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{Filename: viper.GetString("aws-shared-credentials-file"), Profile: viper.GetString("aws-profile")},
```

Comment:
Shouldn't define a const for this strings to avoid typos?

Reply:
Well I could, but then I should also define those constants for all the flags we have, as this is basically the flag name.

Reply:
If they are re-used several times, sounds fine to me :smile: